README.md

```diff
@@ -98,44 +98,95 @@ Usage: gpgbridge.rb [options]
 
 ## Example bash helper functions
 
-I have the following in my `~/.bash_profile` in WSL. Note that there is a
-line in start_gpgbridge that must be uncommented for WSL2.
+Copy the script below to somewhere accessible from WSL, make it 
+executable (`chmod +x path/to/gpgbridge_helper.sh`).
+
+Then add `source path/to/gpgbridge_helper.sh` to your `~/.bash_profile`, `~/.bashrc` or `~/.zshrc`.
+
+Make sure to edit the variables in the start of the script according to your
+setup.
 
 ```bash
+#!/usr/bin/env bash
 #--------------------------------------------------------------------------
 # GPG bridging from WSL gpg to gpg4win gpg-agent.exe
 # (needed to use a Yubikey, since WSL cannot access USB devices)
 
-if [[ -f /mnt/c/Program1/gpgbridge.rb ]]
+# Set this to true if running WSL1
+WSL2=false
+
+SCRIPT_DIR_WSL='/mnt/c/Program1/'
+# shellcheck disable=SC1003
+SCRIPT_DIR_WIN='C:\\Program1\\'
+
+PIDFILE_WSL="$HOME/.gpgbridge.pid"
+LOGFILE_WSL="$HOME/.gpgbridge.log"
+
+PIDFILE_WIN="${SCRIPT_DIR_WIN}gpgbridge.pid"
+LOGFILE_WIN="${SCRIPT_DIR_WIN}gpgbridge.log"
+
+SCRIPT_FILE_NAME='gpgbridge.rb'
+
+SCRIPT_PATH_WSL="${SCRIPT_DIR_WSL}${SCRIPT_FILE_NAME}"
+
+# Set to true to redirect output to /dev/null
+QUIET=false
+
+function start_gpgbridge
+{
+    if ! command -v ruby.exe >/dev/null
+    then
+        echo 'No ruby.exe found in path'
+        return
+    fi
+
+    local cmd=( \
+        'ruby' \
+        "$SCRIPT_PATH_WSL" \
+        '--verbose' \
+        '--daemon' \
+        "--pidfile $PIDFILE_WSL" \
+        "--logfile $LOGFILE_WSL" \
+        "--windows-pidfile $PIDFILE_WIN" \
+        "--windows-logfile $LOGFILE_WIN" \
+    )
+
+    if [ "$1" = 'ssh' ]
+    then
+        cmd+=('--enable-ssh-support')
+        SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket)
+        export SSH_AUTH_SOCK
+    fi
+
+    if [ "$WSL2" = true ]
+    then
+        cmd+=("--remote-address $(ip route | awk '/^default via / {print $3}')")
```

Comment:
Shouldn't this be two separate arguments?

Reply:
It could be, it works as is but that might be more readable?

Reply:
Hmm, maybe it works because of the `eval` line, that will cause all arguments to be split at whitespace.